src/main/java/com/exadel/discountwebapp/location/service/LocationService.java

```diff
@@ -0,0 +1,72 @@
+package com.exadel.discountwebapp.location.service;
+
+import com.exadel.discountwebapp.location.entity.Location;
+import com.exadel.discountwebapp.location.mapper.LocationMapper;
+import com.exadel.discountwebapp.location.repository.LocationRepository;
+import com.exadel.discountwebapp.location.vo.LocationRequestVO;
+import com.exadel.discountwebapp.location.vo.LocationResponseVO;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class LocationService {
+    private final LocationRepository locationRepository;
+    private final LocationMapper locationMapper;
+
+    @Transactional(readOnly = true)
```

Comment:
add please - propagation = Propagation.REQUIRED to be consistent (and below)

Reply:
added